troubleshooting-input-type-file-not-working-on-android.md

```diff
@@ -0,0 +1,89 @@
+slug: troubleshooting-input-type-file-not-working-in-android
+position:
+tags: appbuilder, android
+teampulseid:
+ticketid:  1002377,  1020566, 1018666
+pitsid:
+
+---
+
+## Environment
+<table>
+  <tr>
+    <td>Mobile development type</td>
+    <td>Hybrid (Cordova app)</td>	
+  </tr>
+  <tr>
+    <td>Mobile OS</td>
+    <td>Android</td>	
+  </tr>
+  <tr>
+    <td>Mobile OS version</td>
+    <td>4.4 KitKat and above</td>	
+  </tr>  
+</table>
+
+## Description
+
+Using `<input type="file">`
+doesn't work on some Android devices. In most of the cases, when the **Choose File** button is clicked\tapped there is no response from the web view and the user cannot select a file from the device. 
+
+
+## Steps to Reproduce
+
+1. Create a Cordova application.
+1. Insert the `<input type="file">` element in the `<body>` of the *index.html* file.
+1. Build and Deploy the project to an Android 4.4 or higher device.
+1. Try to interact with the **Choose File** button.
+
+## Cause
+The issue resides in the Chromium web view and as such, it is a bug that Google should address. 
+
+## Solution
+{% include _troubleshooting/no-solution-available.html %}
+
+## Workaround
+The best workaround would be to replace the default web view in your application with Crosswalk, as the input type="file" problem is fixed there. More about Crosswalk can be found [here](https://crosswalk-project.org/).
+
+To do this in Telerik Platform, you can follow the next steps:
+
+1. Open you project in **Code** (formerly AppBuilder).
+1. Make sure that it targets Cordova 4.0 or higher. If it doesn't, you will need to migrate as explained [here](http://docs.telerik.com/platform/appbuilder/cordova/configuring-your-app/configure-target-framework#procedure).
+1. Change the Android web view to Crosswalk as shown [here](http://docs.telerik.com/platform/appbuilder/cordova/configuring-your-app/configure-web-views).
+1. Build and deploy your AppBuilder project on Android device in order to verify that the change resolves the issue.
+
+## Workaround
+Another possible approach would be to use third-party Cordova plugin to handle the file upload on Android. Such plugin can be found [here](https://github.com/MaginSoft/MFileChooser) and to import in your AppBuilder project, you can follow [these steps](http://docs.telerik.com/platform/appbuilder/cordova/using-plugins/using-custom-plugins/add-custom-plugins). 
+
+## Workaround
+If you want to upload an image from the device gallery or take a photo with the device camera, you can also use the Cordova Camera plugin integrated in Telerik Platform. Here are the general steps for this:
+
+1. Enable the Camera plugin from the project's Properties > Plugins page. This is better explained [here](http://docs.telerik.com/platform/appbuilder/cordova/using-plugins/using-core-plugins/enable-disable-core)
+1. In your application, replace the `<input type="file">` element with a call to the Camera plugin API, like this:
+
+```javascript
```

Comment:
Make this JavaScript because the spelling will be preserved in the tab which appears above the code are in the output.